sycl/include/CL/sycl/detail/array.hpp

```diff
@@ -9,6 +9,8 @@
 #pragma once
 #include <CL/sycl/detail/type_traits.hpp>
 #include <CL/sycl/exception.hpp>
+#include <pi/pi.h>
```

Comment:
It's a pity that we had to include `pi.h` here to use `PI_INVALID_VALUE` in the function that has no dependency on plug-in interface.
```c++
  __SYCL_ALWAYS_INLINE void check_dimension(int dimension) const {
#ifndef __SYCL_DEVICE_ONLY__
    if (dimension >= dimensions || dimension < 0) {
      throw cl::sycl::invalid_parameter_error("Index out of range",
                                              PI_INVALID_VALUE);
    }
#endif
    (void)dimension;
  }
```
@romanovvlad, @smaslov-intel, I think there should be PI-independent error codes to report errors for cases like this.

Reply:
Such error codes getting introduced in SYCL2020